neqo-transport/src/server.rs

```diff
@@ -578,6 +588,18 @@ impl Server {
                 packet.wire_version(),
                 self.conn_params.get_versions().all(),
             );
+
+            crate::qlog::server_version_information_failed(
+                &mut self.create_qlog_trace(&AttemptKey {
+                    remote_address: dgram.source(),
```

Comment:
Note that `remote_address` is not used by `self.create_qlog_trace`. Want me to change the signature of `self.create_qlog_trace` to only take the `odcid`?

Reply:
That's worth doing.  We might have needed an address once upon a time.  Or maybe all the other instances where a trace was needed involved an address.  Now that we have a case where  we make a useless `AttemptKey`, it seems worthwhile changing the signature to eliminate that extra work (and have the other call sites use `AttemptKey::odcid()`.

Reply:
:+1: Continuing discussion in https://github.com/mozilla/neqo/pull/1531#pullrequestreview-1823048224.